modules/history/src/views/full/message.jsx

```diff
@@ -0,0 +1,170 @@
+
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+
+import classnames from 'classnames'
+import inspectorTheme from './inspectortheme'
+import JSONTree from 'react-json-tree'
+
+function MessageGroup(props) {
+  const messages = [...props.messages]
+  if (!messages) {
+    return null
+  }
+
+  const userMessageIndex = messages.findIndex(m => m.direction === 'incoming')
+  const userMessage = messages[userMessageIndex]
+  if (userMessage) {
+    messages.splice(userMessageIndex, 1)
+  }
+
+  return (
+    <div className={style['message-group']}>
+      <div className={style['message-group-header']}>
+        {userMessage &&
+          userMessage.decision && (
+            <div className={style['message-group-explanation']}>
+              <div className={style['message-group-confidence']}>{`${Math.round(
+                userMessage.decision.confidence * 10000
+              ) / 100}% decision:`}</div>
+              <div className={style['message-group-decision']}>{` ${userMessage.decision.sourceDetails}`}</div>
+            </div>
+          )}
+        <div className={style['message-inspect']} onClick={() => props.focusMessage(userMessage)}>
+          <MdSearch />
+        </div>
+      </div>
+      <div className={style['message-sender']}>User:</div>
+      {userMessage && (
+        <div className={classnames(style['message-elements'], style['message-incomming'])}>{userMessage.preview}</div>
+      )}
+      <div className={style['message-sender']}>Bot:</div>
+      {messages.map(m => {
+        return (
+          <div
+            className={classnames(
+              style['message-elements'],
+              m.direction === 'outgoing' ? style['message-outgoing'] : style['message-incomming']
+            )}
+            key={`${m.id}: ${m.direction}`}
+            value={m.id}
+          >
+            {m.preview}
+          </div>
+        )
+      })}
+    </div>
+  )
+}
+
+function NoConversationSelected() {
+  return (
+    <div className={style['message-list']}>
+      <div className={style['no-conv']}>
+        <h3>No conversations selected</h3>
+        <p>
+          Please select a conversation on the left pane to see a message history. If there are no conversations
+          available, try talking to your bot and refresh conversations by clicking on the round arrow
+        </p>
+      </div>
+    </div>
+  )
+}
+
+export class MessagesViewer extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      inspectorIsShown: false,
+      currentlyFocusedMessage: null
+    }
+  }
+
+  getLastMessageDate = messageGroups => {
+    const messages = messageGroups.flatMap(m => m)
+    const maxDateMessage = _.maxBy(messages, m => m.createdOn)
+    return new Date(maxDateMessage.createdOn)
+  }
+
+  closeInspector = () => {
+    this.setState({ inspectorIsShown: false })
+  }
+
+  focusMessage(m) {
+    this.setState({ currentlyFocusedMessage: m, inspectorIsShown: true })
+  }
+
```

Comment:
You can use the same syntax as the inspector and remove the .bind(this). Also, I would shorten the name a little:
Eg: `focusMessage = focusedMessage => this.setState({ focusedMessage, isInspectorVisible: true }`

With boolean variable, try to add the conditional string at the beginning (eg: isInspectorVisible, hasBeenDefined, isThatDisplayed, etc)